docs/source/architecture/slashing.rst

```diff
@@ -0,0 +1,72 @@
+The Slashing Protocol
+=====================
+
+TBD
+
+
+Violations
+----------
+
+TBD
+
+
+Calculating the slashing penalty
+--------------------------------
+
+TBD (https://github.com/nucypher/nucypher/issues/803)
+
+
+How slashing affects stake
+--------------------------
+
+The punishment is to reduce the number of tokens that belongs to the offending staker.
+In this case, the main task is not to violate the logic of locking tokens.
+The entire stake consists of:
+
+* tokens which the staker can withdraw at any moment
+* tokens locked for a specific period
+
+Since a staker can divide a stake (extend a part of a stake), as well as lock new tokens, then the total stake is represented as the sum of all the sub stakes active at a particular period. Each sub stake has a beginning and duration, which is reduced only upon confirmation of the activity. Sub stakes get slashed in order based on their duration, beginning with the shortest. Therefore, to start, the unlocked part of tokens belonging to the staker is slashed. After that, if necessary, sub stakes are decreased based on duration: the shortest sub stake is decreased by the required amount; if the adjustment of the sub stake is not enough, then the next shortest sub stake is decreased, and so on. A sub stake that begins in the next period is checked separately.
+
+**Example:**
```

Comment:
I'd love to see some ascii art in here :-)

Reply:
👍

Reply:
YES.